lib/rucio/core/did_meta_plugins/filter_engine.py

```diff
@@ -235,72 +214,121 @@ def _translate_filters(self, model_class, strict_coerce=True):
                 if not isinstance(key_no_suffix, str):
                     mandatory_model_attributes.add(key_no_suffix)
 
-                # logic for value
+                # VALUE
+                # Typecasting is required when the entry point is the CLI as values will always be string.
                 if isinstance(value, str):
-                    value = value.replace('true', 'True').replace('false', 'False')
-                    for format in valid_date_formats:   # try parsing multiple date formats.
-                        try:
-                            value = datetime.strptime(value, format)
-                            break
-                        except ValueError:
-                            continue
-                    try:
-                        value = ast.literal_eval(value)
-                    except (ValueError, SyntaxError):
-                        pass
+                    value = self._try_typecast_string(value)
 
+                # Convert string operator to pythonic operator.
                 and_group_parsed.append(
-                    (key_no_suffix, operators_conversion_LUT.get(oper), value))
+                    (key_no_suffix, OPERATORS_CONVERSION_LUT.get(oper), value))
             filters_translated.append(and_group_parsed)
         self._filters = filters_translated
         return list(mandatory_model_attributes)
 
+    def _try_typecast_string(self, value):
+        """
+        Check if string can be typecasted to bool, datetime or float.
+
+        :param value: The value to be typecasted.
+        :returns: The typecasted value.
+        """
+        value = value.replace('true', 'True').replace('TRUE', 'True')
+        value = value.replace('false', 'False').replace('FALSE', 'False')
+        for format in VALID_DATE_FORMATS:       # try parsing multiple date formats.
+            try:
+                value = datetime.strptime(value, format)
+            except ValueError:
+                continue
+            else:
+                return value
+        try:
+            value = ast.literal_eval(value)     # will catch float, int and bool
+        except (ValueError, SyntaxError):
+            pass
+        return value
+
```

Comment:
When user inputs a metadata filter via CLI, need some way of attempting to cast the value of each key from string (important later when constructing the sqla filter expression).